src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -514,11 +529,23 @@ public void Dispose()
         /// <param name="disposing">If 'true' this method is called from user code; if 'false' it is called by the runtime.</param>
         protected virtual void Dispose(bool disposing)
         {
-            HttpResponseMessage response = _httpResponseMessage;
-            _httpResponseMessage = null;
-            if (response != null)
+            if (_disposed)
             {
-                ((IDisposable)response).Dispose();
+                return;
+            }
+
+            _disposed = true;
+
+            if (disposing)
+            {
+                HttpResponseMessage response = _httpResponseMessage;
+                _httpResponseMessage = null;
+                if (response != null)
+                {
+                    ((IDisposable)response).Dispose();

```

Comment:
I have a feeling the cast here gets JITed away anyway, but I'm curious where this comes from. It's in the original code too. I've only seem something like this with an explicit interface implementation or something. Do you know why we are doing this cast?